docs/reference/plugins/developing_custom_plugins.md

```diff
@@ -0,0 +1,82 @@
+
+```python
+  class MainWidget(QtWidgets.QWidget):
+      """Main plugin widget.
+
+      @param parent shift.ui.widgets.main.ShiftWindow: The Shift main window.
+
+      """
+      def __init__(self, parent=None):
+          super(MainWidget, self).__init__(parent)
+
+          #Store the ShiftWindow inside a class property
+          self.mainWindow = parent
+```
+
+- Wrap all the Qt signal connection calls to the board in a dedicated method (see the `MainWidget._connectBoardSignals` in the *pluginTemplate.py* example file). As each workflow board propagates its own Qt signals, connections must be performed for all existing boards and new ones.
+
+```python
+  def _connectBoardSignals(self):
+      """Standard method to connect to the current board signals."""
+
+      # The signals to be connected from the board...e.g.
+      self.mainWindow.getActiveBoard().workflowExecuted.connect(self._updateContent)
+```
+
+This method should be run for all existing and new board tabs for the Plugin to correctly interact with Shift. The ideal way to do that is to use this method as a Qt slot and connect it to the following signals at the end of the Plugin class constructor logic:
+
+```python
+  def __init__(self, parent=None):
+
+      [...]
+
+      self.mainWindow._boardTabsWidget.currentChanged.connect(self._connectBoardSignals)
+      self.mainWindow.newBoardCreated.connect(self._connectBoardSignals)
+      self._connectBoardSignals()
+```
+
+This will automatically connect the board signals to the currently active one and brand new ones.
+
+## Adding the Plugin to Shift
+
+The first step to add the Plugin to the Shift UI is to create a *json* file containing the information needed by Shift to source all of the custom Plugin classes. This file should be formatted in the following way:
+
+```json
+{
+  "Operators List": {
+    "menu": "Tools",
+    "path": "./operatorsList.py",
+    "class": "OperatorsList"
+  },
+  "Variables Browser": {
+    "menu": "Tools",
+    "path": "./variablesBrowser.py",
+    "class": "VariablesBrowser"
+  }
+}
+```
+
+Each Plugin main entry name will define the name displayed by the Plugin menu entry in the top bar menu. Then for each Plugin entry will need:
+
+- The name of the top menu entry that the Plugin should be added to.
+- The absolute or relative path to the Python file containing the Plugin class.
+- The name of the Plugin class to be imported.
+
+Shift allows users to add their custom plugins even to native menus. If a new menu name is requested, it will be added to the UI automatically.
+
+Once the .json file is created and populated, the Plugin can be sourced by appending the path to the *plugins.json* file to the `SHIFT_PLUGIN_PATH` environment variable before starting Shift:
```

Comment:
To be aligned with the paragraph at line 55:
```suggestion
Once the *json* file is created and populated, the Plugin can be sourced by appending the path to the *plugins.json* file to the `SHIFT_PLUGIN_PATH` environment variable before starting Shift:
```

Reply:
done